core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/CodeSystemEntry.java

```diff
@@ -340,4 +346,28 @@ public boolean equals(final Object obj) {
 			return false;
 		return true;
 	}
-}
\ No newline at end of file
+	
+	/**
+	 * Returns all code system short name dependencies and itself.
+	 */
+	@JsonIgnore
+	public SortedSet<String> getDependenciesAndSelf() {
+		Set<String> affectedCodeSystems = Sets.newHashSet(CoreTerminologyBroker.getInstance().getAffectedCodeSystemsForTeminology(terminologyComponentId));
```

Comment:
🛑 This method still creates too many unnecessary objects to return a `SortedSet` of Strings.